File: src/main/java/co/uk/jiveelection/campaign/input/Input.java
package co.uk.jiveelection.campaign.input;

import co.uk.jiveelection.campaign.output.twitter.TranslationEntity;

import java.util.List;

/**
 * Inputs non Jive strings.
 */
public interface Input<I> {
    List<TranslationEntity> extractEntities(I input);
}

File: src/main/kotlin/co/uk/jiveelection/campaign/input/Input.kt
package co.uk.jiveelection.campaign.input

import co.uk.jiveelection.campaign.output.twitter.TranslationEntity

/**
 * Inputs non Jive strings.
 */
interface Input<in I> {
    fun extractEntities(input: I): List<TranslationEntity>
}
